src/ui/RecipeForm.tsx

```diff
@@ -42,61 +60,66 @@ const RecipeForm: FC = () => {
         ingredients: [{ name: '', qty: '', units: '' }],
         directions: [''],
         notes: '',
-        file: null,
+        file: {},
         email: '',
         contact: false,
         reviewed: false,
         hidden: false,
       }}
-      onSubmit={(values) => console.log(values, null, 2)}
+      validationSchema={formValueSchema}
+      onSubmit={async (values) => {
+        const parsedValues = await formValueSchema.validate(values);
+        const response = await recipeService.create(parsedValues);
+        goToRecipePage(response.id);
+      }}
     >
-      {({ handleSubmit, values }) => {
+      {({ handleSubmit }) => {
         return (
           <form onSubmit={handleSubmit}>
+            <ErrorLabel name="recipeName" />
             <Input name="recipeName" placeholder="What is the name of the dish?" />
+            <ErrorLabel name="category" />
             <Select
               name="category"
               options={categories}
               placeholder="What kind of recipe is this?"
             />
+            <ErrorLabel name="author" />
             <Input name="author" placeholder="Tell us your name..." />
+            <ErrorLabel name="bydel" />
             <Select name="bydel" options={bydels} placeholder="Select your area" />
+            <ErrorLabel name="story" />
             <TextArea name="story" placeholder="What makes this dish special to you?" />
+            <ErrorLabel name="yield" />
             <Select
               name="yield"
               placeholder="How many servings does this recipe produce?"
-              // Create an array containing ints 0..N, remove 0, concat an 'N+' string as final element
               options={Array(12)
                 .fill(true)
                 .map((_, i) => (i === 11 ? `${i + 1}+` : `${i + 1}`))}
             />
+            <ErrorLabel name="prepTime" />
             <InputPrepTime name="prepTime" />
+            <ErrorLabel name={`ingredients.0.name`} />
             <Ingredients />
             <DirectionsBox />
+            <ErrorLabel name="notes" />
             <TextArea name="notes" placeholder="Any additional notes go here!" />
             {/* <input
-                id="file"
-                name="file"
-                type="file"
-                onChange={(event) => {
-                  props.setFieldValue('file', event.currentTarget.files[0]);
-                }}
-              /> */}
+              id="file"
+              name="file"
+              type="file"
+              onChange={(event) => {
+                props.setFieldValue('file', event.currentTarget.files[0]);
+              }}
+            /> */}
+            <ErrorLabel name="email" />
             <Input name="email" placeholder="Email" />{' '}
             <label>
               <Field type="checkbox" name="contact" /> Would you be open to sharing the story behind
```

Comment:
Feel like everything in this label component should be separate component. Or a reusable `<Checkbox />` component